### DISCUSSION TITLE
Unable to Install on macOS Ventura - Portable Ruby Install Fails

### DISCUSSION THREAD

**Discussion by OldCrow** [Homebrew/discussions#5660] [Getting started] [answered]
### Output of `brew config`

```shell
N/A - new installation
```

### Output of `brew doctor`

```shell
N/A - new installation
```

### Description of issue

Attempting to install on a "new" 2017 MBP running macOS Ventura 13.7. Installation fails. Command line output follows:

% /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
==> Checking for `sudo` access (which may request your password)...
Password:
==> This script will install:
/usr/local/bin/brew
/usr/local/share/doc/homebrew
/usr/local/share/man/man1/brew.1
/usr/local/share/zsh/site-functions/_brew
/usr/local/etc/bash_completion.d/brew
/usr/local/Homebrew
==> The following new directories will be created:
/usr/local/etc
/usr/local/include
/usr/local/lib
/usr/local/sbin
/usr/local/share
/usr/local/var
/usr/local/opt
/usr/local/share/zsh
/usr/local/share/zsh/site-functions
/usr/local/var/homebrew
/usr/local/var/homebrew/linked
/usr/local/Cellar
/usr/local/Caskroom
/usr/local/Frameworks

Press RETURN/ENTER to continue or any other key to abort:
==> /usr/bin/sudo /bin/mkdir -p /usr/local/etc /usr/local/include /usr/local/lib /usr/local/sbin /usr/local/share /usr/local/var /usr/local/opt /usr/local/share/zsh /usr/local/share/zsh/site-functions /usr/local/var/homebrew /usr/local/var/homebrew/linked /usr/local/Cellar /usr/local/Caskroom /usr/local/Frameworks
==> /usr/bin/sudo /bin/chmod ug=rwx /usr/local/etc /usr/local/include /usr/local/lib /usr/local/sbin /usr/local/share /usr/local/var /usr/local/opt /usr/local/share/zsh /usr/local/share/zsh/site-functions /usr/local/var/homebrew /usr/local/var/homebrew/linked /usr/local/Cellar /usr/local/Caskroom /usr/local/Frameworks
==> /usr/bin/sudo /bin/chmod go-w /usr/local/share/zsh /usr/local/share/zsh/site-functions
==> /usr/bin/sudo /usr/sbin/chown wolfman /usr/local/etc /usr/local/include /usr/local/lib /usr/local/sbin /usr/local/share /usr/local/var /usr/local/opt /usr/local/share/zsh /usr/local/share/zsh/site-functions /usr/local/var/homebrew /usr/local/var/homebrew/linked /usr/local/Cellar /usr/local/Caskroom /usr/local/Frameworks
==> /usr/bin/sudo /usr/bin/chgrp admin /usr/local/etc /usr/local/include /usr/local/lib /usr/local/sbin /usr/local/share /usr/local/var /usr/local/opt /usr/local/share/zsh /usr/local/share/zsh/site-functions /usr/local/var/homebrew /usr/local/var/homebrew/linked /usr/local/Cellar /usr/local/Caskroom /usr/local/Frameworks
==> /usr/bin/sudo /bin/mkdir -p /usr/local/Homebrew
==> /usr/bin/sudo /usr/sbin/chown -R wolfman:admin /usr/local/Homebrew
==> /usr/bin/sudo /bin/mkdir -p /Users/wolfman/Library/Caches/Homebrew
==> /usr/bin/sudo /bin/chmod g+rwx /Users/wolfman/Library/Caches/Homebrew
==> /usr/bin/sudo /usr/sbin/chown -R wolfman /Users/wolfman/Library/Caches/Homebrew
==> Downloading and installing Homebrew...
remote: Enumerating objects: 283529, done.
remote: Counting objects: 100% (18555/18555), done.
remote: Compressing objects: 100% (789/789), done.
remote: Total 283529 (delta 17934), reused 18206 (delta 17728), pack-reused 264974 (from 1)
remote: Enumerating objects: 55, done.
remote: Counting objects: 100% (34/34), done.
remote: Total 55 (delta 34), reused 34 (delta 34), pack-reused 21 (from 1)
==> Updating Homebrew...
==> Downloading https://ghcr.io/v2/homebrew/portable-ruby/portable-ruby/blobs/sha256:e02b387d80f10c835df15115360b0b5deb8e35f8967c7e68c9942af046023209
################################################################################################################# 100.0%
==> Pouring portable-ruby-3.3.5.el_capitan.bottle.tar.gz
/usr/local/Homebrew/Library/Homebrew/cmd/vendor-install.sh: line 229: 26427 Killed: 9               "./${VENDOR_VERSION}/bin/${VENDOR_NAME}" --version > /dev/null
Error: Failed to install ruby 3.3.5!
Error: Failed to install Homebrew Portable Ruby (and your system version is too old)!
Failed during: /usr/local/bin/brew update --force --quiet

As can be seen, everyhting works as expected right up until the Portable Ruby 3.3.5 bottle is meant to be poured. Not sure why it is pouring an El Capitan bottle, though, as El Capitan had some fixes to get around pre-exec crash issues - it this perhaps a build script error and the wrong bottle is being poured?

In any case, as Ventura is still an Apple-supported macOS version, Homebrew should still work "out of the box" as it were.

Any assistance would be apprecaited.

**Comment by gromgit** [comment]
> Not sure why it is pouring an El Capitan bottle, though

That seems to be hardcoded for Intel Darwin installs:
```
$ cat /usr/local/Library/Homebrew/vendor/portable-ruby-x86_64-darwin
ruby_TAG=el_capitan
ruby_SHA=e02b387d80f10c835df15115360b0b5deb8e35f8967c7e68c9942af046023209
```

What's the output of `/usr/local/Library/Homebrew/vendor/portable-ruby/3.3.5/bin/ruby --version`?

**Comment by p-linnane** [comment]
Portable Ruby is built for the oldest system we support for each architecture. This is El Capitan on x86_64, and Big Sur on arm64. I'm not sure what is causing your issue though. @Bo98 any ideas?

**Comment by Bo98** [comment, marked as answer]
Every single report about `Killed: 9` we've received so far has been because of third-party security software blocking the install (e.g. software that blocks unsigned CLI tools or intercepts everything downloaded via `curl` and "quarantines" it like a Safari download), so that's the first thing to check.

**Reply by OldCrow** [reply]
Hmm, interesting. Wouldn't think that was the issue since I have the same tools on this laptop as on my M1 Mini running Sequoia, and that updated Portable Ruby just fine (even if the Intel and ARM Portable Ruby versions are different).

Still, I might have missed a modal dialogue or something at some point and failed to authorize something allowing iTerm to run the CLI tool properly. I'll kill the security tools and see if I can make it work later and report back.

**Reply by OldCrow** [reply]
Turned out to be Vallum Endpoint Security - not sure what I configured differently in my two systems, but that's a tomorrow problem to figure out
